### ISSUE TITLE
VoiceOver issues with navigation-side-menu

### ISSUE THREAD

**Issue by bjankord** [cerner/terra-framework#284]
# Bug Report

## Description
When using VoiceOver on iOS, no indication was given of which item was selected in the [side navigation menu example](https://engineering.cerner.com/terra-framework/#/components/terra-navigation-side-menu/navigation-side-menu/navigation-side-menu). 
 VO also did not indicate the menu items where interactable (a button or a link).

## Steps to Reproduce

1. Navigate to https://engineering.cerner.com/terra-framework/#/components/terra-navigation-side-menu/navigation-side-menu/navigation-side-menu
2. Interact with side navigtion menu with VoiceOver

## Expected Behavior
* Selected side navigation menu item is communicated to screen reader users.
* Screen reader users are made aware of interactable elements in side navigation.

## Environment

* Component Name and Version: terra-navigation-side-menu

**Comment by foxannefoxanne** [comment]
Is there any progress on this issue? Our app recently did an accessibility audit and notices this as a large fail point. It was also noted that there was not a great indicator of sub navigation items, as well as no great indicator that you had scroll through the entire menu and were back to main content (may have  been resolved with the recent Aria landmark changes?)

**Comment by ryanthemanuel** [comment]
@foxannefoxanne we currently have someone working on this.  Hopefully it'll be fixed this week.

**Comment by bjankord** [comment]
@foxannefoxanne We added aria landmark roles for navigation and main that should better indicate where the user is at in the page (DOM). We also resolved the following issues with #761.
* Selected side navigation menu item is communicated to screen reader users.
* Screen reader users are made aware of interact-able elements in side navigation.

> It was also noted that there was not a great indicator of sub navigation items

I'll log a new issue to track this enhancement. We need to get translations for this but it should be fairly straight forward to get this indicator added for sub navigation.
